app/src/main/java/de/westnordost/streetcomplete/quests/localized_name/data/RoadNameSuggestionsDao.kt

```diff
@@ -74,4 +78,27 @@ class RoadNameSuggestionsDao @Inject constructor(
         }
         return result
     }
+
+    fun putRoadNameSuggestion(element: Element, geometry: ElementGeometry?) {
+        if (element.type != Element.Type.WAY) return
+        if (geometry !is ElementPolylinesGeometry) return
+        val namesByLanguage = element.tags?.toRoadNameByLanguage() ?: return
+
+        putRoad(element.id, namesByLanguage, geometry.polylines.first())
+    }
+
+    /** OSM tags (i.e. name:de=Bäckergang) to map of language code -> name (i.e. de=Bäckergang) */
+    private fun Map<String,String>.toRoadNameByLanguage(): Map<String, String>? {
+        val result = mutableMapOf<String,String>()
+        val namePattern = Pattern.compile("name(:(.*))?")
+        for ((key, value) in this) {
+            val m = namePattern.matcher(key)
+            if (m.matches()) {
+                val languageCode = m.group(2) ?: ""
+                result[languageCode] = value
+            }
+        }
+        return if (result.isEmpty()) null else result
+    }
+
```

Comment:
Could this be extension functions in a kotlin file in the same directory as the address quests instead? No need to blow up the interface.

Reply:
This functionality is used by both AddRoadNameForm and AddAddressStreetForm and RoadNameSuggestionDao was already in AddAddressStreet, so I added the functionality here for both forms.